src/samples/Frame.cpp

```diff
@@ -26,13 +26,13 @@ bool frame::frame_size_t::operator!=(const frame::frame_size_t& other) const
     return !(*this == other);
 }
 
-frame::Frame::Frame() : image(), size(), data_depth(0), pixel_size(0), frame_mode()
+frame::Frame::Frame() : image(), size(), data_depth(0), pixel_size(0), frame_mode(MODE_UNDEFINED)
 {
     setDataDepth(0);
     reset();
 }
 
-frame::Frame::Frame(uint16_t width, uint16_t height, uint8_t depth, frame::frame_mode_t mode, const uint8_t val, size_t sizeInBytes)
+frame::Frame::Frame(uint16_t width, uint16_t height, uint8_t depth, frame::frame_mode_t mode, int16_t val, size_t sizeInBytes)
```

Comment:
I would make `val` an `int`/`int32_t` in case someone wants to initialize a Frame with `depth=16` by `0xffff` (may not happen often, but won't cost anything)
(same for other occurrences of `val`)

Reply:
The code currently does not handle that case (it's simply filling the vector which is a int8_t)